### PR TITLE
opt: taskpool move to endpoints layer

### PR DESCRIPTION
**Pull request by watermelo**

### CODE REVIEW

**Review by AlexStocks — APPROVED** [review]

**Comment by AlexStocks** [comment]
Pls update gost version to v1.9.3 to enhance task pool robustness.

**Review by AlexStocks — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

File: transport/session.go [hunk 1]
```diff
@@ -324,7 +324,7 @@ func (s *session) SetWaitTime(waitTime time.Duration) {
 	s.wait = waitTime
 }
 
-// set task pool
+// Deprecated: set task pool
```
Comment (wongoo) [inline transport/session.go:327]: the func inner implement can be removed, just leave a empty func for compatibility

File: transport/options.go [hunk 2]
```diff
@@ -83,6 +94,8 @@ type ClientOptions struct {
 	// duration, the hash alg, the len of the private key.
 	// wss client will use it.
 	cert string
+	// task queue
```
Comment (wongoo) [inline transport/options.go:97]: task pool

File: transport/options.go [hunk 3]
```diff
@@ -31,6 +33,8 @@ type ServerOptions struct {
 	cert       string
 	privateKey string
 	caCert     string
+	// task queue
```
Comment (wongoo) [inline transport/options.go:36]: task pool

**Review by wongoo — APPROVED** [review]

**Review by fangyincheng — APPROVED** [review]

File: transport/session.go (hunk 1 above)
Comment (fangyincheng) [reply transport/session.go:327]: an empty func will make users lose original logic.

**Pull request merged**